src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -209,7 +209,28 @@ private static void BindNonScalar(this IConfiguration configuration, object inst
         {
             if (instance != null)
             {
-                foreach (PropertyInfo property in GetAllProperties(instance.GetType()))
+                List<PropertyInfo> modelProperties = GetAllProperties(instance.GetType());
+
+                if (options.ErrorOnUnknownConfiguration)
+                {
+                    HashSet<string> propertyNames = new(modelProperties.Select(mp => mp.Name),
+                        StringComparer.OrdinalIgnoreCase);
+
+                    IEnumerable<IConfigurationSection> configurationSections = configuration.GetChildren().ToList();
```

Comment:
Do you need `ToList` here? I doesn't look like it.

Reply:
Well spotted.  No, we don't.  Would you like me to change it?

Reply:
I'll include it here to make it easer. 
```suggestion
                    IEnumerable<IConfigurationSection> configurationSections = configuration.GetChildren();
```